hoodie-client/src/test/java/com/uber/hoodie/common/HoodieClientTestUtils.java

```diff
@@ -173,4 +180,59 @@ public static SparkConf getSparkConfForTest(String appName) {
       throw new HoodieException("Error reading hoodie dataset as a dataframe", e);
     }
   }
+
+  public static String writeParquetFile(String basePath,
+                                        String partitionPath,
+                                        String filename,
+                                        List<HoodieRecord> records,
+                                        Schema schema,
+                                        BloomFilter filter,
+                                        boolean createCommitTime) throws IOException {
+
+    if (filter == null) {
+      filter = new BloomFilter(10000, 0.0000001);
+    }
+    HoodieAvroWriteSupport writeSupport = new HoodieAvroWriteSupport(new AvroSchemaConverter().convert(schema), schema,
+        filter);
+    String commitTime = FSUtils.getCommitTime(filename);
+    HoodieParquetConfig config = new HoodieParquetConfig(writeSupport, CompressionCodecName.GZIP,
+        ParquetWriter.DEFAULT_BLOCK_SIZE, ParquetWriter.DEFAULT_PAGE_SIZE, 120 * 1024 * 1024,
+        HoodieTestUtils.getDefaultHadoopConf(),
+        Double.valueOf(HoodieStorageConfig.DEFAULT_STREAM_COMPRESSION_RATIO));
+    HoodieParquetWriter writer = new HoodieParquetWriter(
+        commitTime,
+        new Path(basePath + "/" + partitionPath + "/" + filename),
+        config,
+        schema);
+    int seqId = 1;
+    for (HoodieRecord record : records) {
+      GenericRecord avroRecord = (GenericRecord) record.getData().getInsertValue(schema).get();
+      HoodieAvroUtils.addCommitMetadataToRecord(avroRecord, commitTime, "" + seqId++);
+      HoodieAvroUtils.addHoodieKeyToRecord(avroRecord, record.getRecordKey(), record.getPartitionPath(), filename);
+      writer.writeAvro(record.getRecordKey(), avroRecord);
+      filter.add(record.getRecordKey());
+    }
+    writer.close();
+
+    if (createCommitTime) {
+      HoodieTestUtils.createMetadataFolder(basePath);
+      HoodieTestUtils.createCommitFiles(basePath, commitTime);
+    }
+    return filename;
+  }
+
+  public static String writeParquetFile(String basePath,
+                                        String partitionPath,
+                                        List<HoodieRecord> records,
+                                        Schema schema,
+                                        BloomFilter filter,
+                                        boolean createCommitTime) throws IOException, InterruptedException {
+    Thread.sleep(1000);
```

Comment:
@leletan any reason for the sleep?

Reply:
@vinothchandar nope, this is just copying from https://github.com/uber/hudi/blob/master/hoodie-client/src/test/java/com/uber/hoodie/index/bloom/TestHoodieBloomIndex.java#L496

Reply:
actually the 2 parquet writing helper functions are mostly just copied from that file